src/main/java/com/teragrep/jai_02/keystore/KeyStoreFactory.java

```diff
@@ -0,0 +1,106 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jai_02.keystore;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+
+/**
+ * Provides a factory, which builds KeyStores from the
+ * defined path and format.
+ */
+public class KeyStoreFactory {
+
+    private final KeyStoreFormat keyStoreFormat;
+    private final String path;
+    private final char[] pw;
+    public KeyStoreFactory() {
+        this(new KeyStoreFormat(), null, null);
+    }
+
+    public KeyStoreFactory(String path, char[] pw) {
+        this(new KeyStoreFormat(), path, pw);
+    }
+
+    public KeyStoreFactory(KeyStoreFormat ka) {
+        this(ka, null, null);
+    }
+
+    public KeyStoreFactory(KeyStoreFormat ka, String path, char[] pw) {
+        this.keyStoreFormat = ka;
+        this.path = path;
+        this.pw = pw;
+    }
+
+    public KeyStore build() {
+        final KeyStore ks;
+        try {
+            ks = KeyStore.getInstance(keyStoreFormat.toString());
+            if (path == null || Files.notExists(Paths.get(path), LinkOption.NOFOLLOW_LINKS)) {
```

Comment:
Is this part used in "getOrCreate(path)" style and the path will be populated later? Otherwise I'd treat provided but missing file as a failure rather than silently fallback to using nulls

Reply:
changed to throw exception if no path given, but if path given and keyStore file does not exist, create it